Stratis.Bitcoin/Features/RPC/Controllers/FullNodeController.cs

```diff
@@ -64,10 +56,14 @@ public Task Stop()
             if (!uint256.TryParse(txid, out trxid))
                 throw new ArgumentException(nameof(txid));
 
-            Transaction trx = (await this.MempoolManager?.InfoAsync(trxid))?.Trx;
+            var transactionProvider = this.FullNode.Services?.Features.OfType<IPooledTransaction>()?.FirstOrDefault();
+            Transaction trx = await transactionProvider?.GetTransaction(trxid);
 
             if (trx == null)
-                trx = await this.BlockManager?.BlockRepository?.GetTrxAsync(trxid);
+            {
+                var blockStoreProvider = this.FullNode.Services?.Features.OfType<IBlockStore>()?.FirstOrDefault();
```

Comment:
This can be refactored a bit (FYI all features are also registered with the service provider but not the interface)
Maybe add a method so we can do the following
```
this.FullNode.NodeFeature<IPooledTransaction>()
```